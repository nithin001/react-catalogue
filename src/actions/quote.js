import { put } from '../api/cart';
export const load = (cart) => {
  return (dispatch) => {
    dispatch({ type: 'LOAD_QUOTE_START' });
    return put(cart)
      .then((quote) => {
        dispatch({ type: 'POPULATE_QUOTE', payload: quote });
        dispatch({ type: 'LOAD_QUOTE_COMPLETE' });
      })
      .catch(() => {
        dispatch({ type: 'LOAD_ERROR' });
      });
  };
};