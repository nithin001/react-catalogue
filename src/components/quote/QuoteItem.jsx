import React from 'react';

export const QuoteItem = ({ order }) => (
  <div className={'quote-item'}>
    <span className={'sku'}>{order.get('sku')}</span>
    <span className={'name'}>{order.get('name')}</span>
    <span className={'price-amount'}>{order.getIn(['price', 'amount'])}</span>
    <span className={'price-currency'}>{order.getIn(['price', 'currency'])}</span>
    <span className={'quantity'}>{order.get('quantity')}</span>
    <span className={'total-amount'}>{order.getIn(['lineTotal', 'amount'])}</span>
    <span className={'total-currency'}>{order.getIn(['lineTotal', 'currency'])}</span>
  </div>
);