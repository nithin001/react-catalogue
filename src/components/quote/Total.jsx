import React from 'react';

export const Total = ({ total }) => (
  <div className={'quote-total'}>
    <span className={'amount'}>{total.get('amount')}</span>
    <span className={'currency'}>{total.get('currency')}</span>
  </div>
);