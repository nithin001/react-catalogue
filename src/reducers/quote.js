import Immutable from 'immutable';
const quote = (state = Immutable.Map(), action) => {
  if (action.type === 'POPULATE_QUOTE') {
    return Immutable.fromJS(action.payload);
  }
  return state;
};
export default quote;