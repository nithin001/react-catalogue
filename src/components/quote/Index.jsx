import React from 'react';
import { Total } from './Total';
import { QuoteItem } from './QuoteItem';
export const Quote = ({ quote }) => {
  const quoteItems = quote.get('lines').map(quoteItem => {
    return (<QuoteItem key={quoteItem.get('sku')} order={quoteItem}/>);
  });

  return (
    <div className={'quote'}>
      {quoteItems}
      <Total total={quote.get('total')}/>
    </div>
  );
};