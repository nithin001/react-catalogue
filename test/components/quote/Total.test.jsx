import React from 'react';
import renderer from 'react-test-renderer';
import quote from '../../../src/reducers/quote';
import { Total } from '../../../src/components/quote/Total';
var mockQuote = {
  'lines': [{
    'sku': '374847',
    'name': 'Nice Product',
    'price': {
      'amount': 20,
      'currency': 'EUR'
    },
    'quantity': 3,
    'lineTotal': {
      'amount': 60,
      'currency': 'EUR'
    }
  }],
  'total': {
    'amount': 60,
    'currency': 'EUR'
  }
};
const quoteState = quote(undefined, { type: 'POPULATE_QUOTE', payload: mockQuote });
describe('QuoteItem component', () => {
  it('should render the quote item with name, price, quantity and line total', () => {
    const props = {
      total: quoteState.get('total')
    };
    const total = renderer.create(<Total
      {...props}
    />);
    expect(total).toMatchSnapshot();
  });
});