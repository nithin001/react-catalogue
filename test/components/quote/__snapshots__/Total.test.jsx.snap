// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteItem component should render the quote item with name, price, quantity and line total 1`] = `
<div
  className="quote-total"
>
  <span
    className="amount"
  >
    60
  </span>
  <span
    className="currency"
  >
    EUR
  </span>
</div>
`;
